### DISCUSSION TITLE
Verifying instantaneous rather than max field

### DISCUSSION THREAD

**Discussion by jwolff-ncar** [dtcenter/METplus#1151] [File I/O] [answered]
Hi!

I have a dataset than includes 10m MAX wind speed but does not include 10m instantaneous wind speed. 

wgrib2 -V -d 853 rrfs.t00z.bgdawpf006.tm00.grib2
853:3628760795:vt=2019041506:10 m above ground:5-6 hour max fcst:WIND Wind Speed [m/s]:
(there is no 10 m above ground:6 hour fcst:WIND Wind Speed in the file)

But these fields are also available in that same file
847:3607177138:vt=2019041506:10 m above ground:6 hour fcst:UGRD U-Component of Wind [m/s]:
848:3615388742:vt=2019041506:10 m above ground:6 hour fcst:VGRD V-Component of Wind [m/s]:

Is there a way to tell MET to ignore the max field and derive 10m wind using the 10m u/v components for the instantaneous values instead? 

Thanks!

**Comment by JohnHalleyGotway** [comment, marked as answer]
@jwolff-ncar thanks for moving this issue over to Discussions.

As we discussed via email, the short answer is that adding the "GRIB2_pdt = 0;" option to the configuration string produces the desired result. With that option, when you request WIND (for wind speed), MET will find the U and V wind components and derive it for you on the fly. 

In general, when requesting wind speed (WIND) or direction (WDIR) from a GRIB file, MET first tries to find that field directly in the input file. If it's not found, MET automatically searches for the corresponding U and V wind components, rotates them from grid-to-earth relative, if necessary, and derives the requested field.

Note though that this derivation logic has only been tested using GRIB1/2 files produced by NOAA/EMC. The logic depends on the U and V wind components being name "UGRD" and "VGRD" respectively. So it likely will not work well for non-GRIB and non-NOAA GRIB1/2 datasets. If needed, I'm sure we could find a more configurable general solution for this.

For your sample file with hundreds of GRIB2 records, I ran wgrib2 to extract only the 3 GRIB records we're concerned about:
```
wgrib2 rfs.t00z.bgdawpf006.tm00.grib2 | \
   grep ":10 m above" | egrep "UGRD|VGRD|WIND" | \
   wgrib2 -i -grib_out rrfs.t00z.bgdawpf006.tm00.10mWIND.grib2 \
   rrfs.t00z.bgdawpf006.tm00.grib2
```
And here are those 3 records:
```
wgrib2 rrfs.t00z.bgdawpf006.tm00.10mWIND.grib2
1:0:d=2019041500:UGRD:10 m above ground:6 hour fcst:
2:17346776:d=2019041500:VGRD:10 m above ground:6 hour fcst:
3:34693552:d=2019041500:WIND:10 m above ground:5-6 hour max fcst:
```
Note that the first 2 are instantaneous 6-hour forecasts while the 3rd is a 5-6 hour max. Running MET's plot_data_plane tool to plot the wind speed, as below, finds the variable named "WIND" in the 3rd record of that file:
```
plot_data_plane rrfs.t00z.bgdawpf006.tm00.10mWIND.grib2 10mWIND.ps 'name="WIND"; level="Z10";' -v 3
DEBUG 3: MetGrib2DataFile::data_plane() -> Found exact match for 'WIND/Z10' in GRIB2 record 3 field 1 of GRIB2 file 'rrfs.t00z.bgdawpf006.tm00.10mWIND.grib2'
```
Trying to diagnose the difference in "5-6 hour max fcst" vs "6 hour fcst" is harder than it should be. This info is in the GRIB Product Description Section, which is section number 4. Rerun wgrib2 using the "-Sec4" option:
```
wgrib2 -Sec4 rrfs.t00z.bgdawpf006.tm00.10mWIND.grib2
1:0:Sec4 len=34 #vert coordinate=0 Product Defn Template=4.0 size=34 free=0
2:17346776:Sec4 len=34 #vert coordinate=0 Product Defn Template=4.0 size=34 free=0
3:34693552:Sec4 len=58 #vert coordinate=0 Product Defn Template=4.8 size=58 free=0
```
Note that the 3rd record follows template number 4.8 instead of 4.0. In the MET documentation, search for GRIB2_pdt:
https://met.readthedocs.io/en/develop/Users_Guide/config_options.html

Re-running plot_data_plane using "GRIB2_pdt = 0;", MET no longer finds the 3rd record (which has GRIB2_pdt = 8). Since it's not found, it derives wind speed from the U and V components in that file:
```
plot_data_plane rrfs.t00z.bgdawpf006.tm00.10mWIND.grib2 10mWIND-pdt0.ps 'name="WIND"; level="Z10"; GRIB2_pdt=0;' -v 3
DEBUG 3: Deriving wind speed from U and V wind components.
```
Here's the result... original on left and with "GRIB2_pdt = 0" on the right.
<img width="1375" alt="Screen Shot 2021-09-09 at 1 26 08 PM" src="https://user-images.githubusercontent.com/21087144/132882017-498b0e9a-f597-4a38-858b-8bc7b1ea3871.png">
